src/utils/api/withAuthorization.ts

```diff
@@ -0,0 +1,28 @@
+import {
+	APIGatewayProxyEvent,
+	APIGatewayProxyHandler,
+	APIGatewayProxyResult,
+	Callback,
+	Context,
+} from 'aws-lambda';
+import { validateAuthorizationToken } from '@utils/api/apiAuth';
+
+export function withAuthorization(handler: APIGatewayProxyHandler): APIGatewayProxyHandler {
```

Comment:
This is a comment about my preference, but I don't like higher-order functions like this. I think it makes debugging harder as you have to find this and understand how it works. I don't think it's a bad practice per see, just a personal choice. There's some behaviors I'd adjust though so I'll comment on that.